designs/Net5PlatformAndPlatformVersion.md

```diff
@@ -0,0 +1,177 @@
+
+For more details on the feature itself, please refer to [the full spec](https://github.com/terrajobst/designs/blob/master/accepted/2020/net5/net5.md).
+
+## class NuGetFramework
+
+Source: src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs
+
+### Fields
+
+Three additional fields will be added:
+
+```csharp
+        /// <summary>
+        /// Framework Platform (net5.0+)
+        /// </summary>
+        public string Platform
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Framework Platform Version (net5.0+)
+        /// </summary>
+        public Version PlatformVersion
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// True if platform is non-empty (net5.0+)
+        /// </summary>
+        public bool HasPlatform
+        {
+            get { return !string.IsNullOrEmpty(Platform); }
+        }
+
+        /// <summary>
+        /// True if platform version is non-empty (net5.0+)
+        /// </summary>
+        public bool HasPlatformVersion
+        {
+            get { return IsNet5Era && !(PlatformVersion.Major == 0 && PlatformVersion.Minor == 0); }
+        }
+```
+
+### Constructors
+
+An additional constructor should be added that accepts `Version platformVersion` as its 4th argument. The current code in `NuGetFramework/3` should be moved to this new constructor, and `string profile` should be renamed to `string profileOrPlatform`. This field will be EITHER a legacy profile or, if net5.0+, a platform identifier.
+
+```csharp
+        public NuGetFramework(string framework, Version version, string profileOrPlatform)
+            : this(framework, version, profileOrPlatform, FrameworkConstants.EmptyVersion)
+        {
+        }
+
+        private const int Version5 = 5;
+
+        public NuGetFramework(string frameworkIdentifier, Version frameworkVersion, string profileOrPlatform, Version platformVersion)
+        {
+            if (frameworkIdentifier == null)
+            {
+                throw new ArgumentNullException("frameworkIdentifier");
+            }
+
+            if (frameworkVersion == null)
+            {
+                throw new ArgumentNullException("frameworkVersion");
+            }
+
+            _frameworkIdentifier = frameworkIdentifier;
+            _frameworkVersion = NormalizeVersion(frameworkVersion);
+
+            IsNet4Era = (_frameworkVersion.Major >= Version5 && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, _frameworkIdentifier));
+
+            _frameworkProfile = IsNet5Era ? profileOrPlatform ?? string.Empty : string.Empty;
+            Platform = IsNet5Era ? profileOrPlatform ?? string.Empty : string.Empty;
+            PlatformVersion = NormalizeVersion(platformVersion ?? FrameworkConstants.EmptyVersion);

```

Comment:
why empty values to represent values that aren't defined?  It's a perfect use-case for null. I feel like I'm missing information about why `_frameworkProfile` already does this.

Reply:
I agree, but this is what @nkolev92 pushed for, and I ended up conceding. Happy to jump on a call if you want to bring back the discussion. fwiw, I agree that this is a good use-case for null, in general.

Reply:
For reference :) 
https://github.com/NuGet/Home/pull/9382#discussion_r404417914